merge.go

```diff
@@ -0,0 +1,157 @@
+package zoekt
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+func Merge(dstDir string, files ...IndexFile) error {
+	var ds []*indexData
+	for _, f := range files {
+		searcher, err := NewSearcher(f)
+		if err != nil {
+			return err
+		}
+		ds = append(ds, searcher.(*indexData))
+	}
+
+	ib, err := merge(ds...)
+	if err != nil {
+		return err
+	}
+
+	hasher := sha1.New()
+	for _, d := range ds {
+		for i, md := range d.repoMetaData {
+			if d.repoTombstone[i] {
+				continue
+			}
+			hasher.Write([]byte(md.Name))
+			hasher.Write([]byte{0})
+		}
+	}
+
+	fn := filepath.Join(dstDir, fmt.Sprintf("compound-%x_v%d.%05d.zoekt", hasher.Sum(nil), IndexFormatVersion, 0))
```

Comment:
Are we writing out v16 for compound shards?

Reply:
fixed